sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -62,9 +63,59 @@
     "blob": {"primary": "BlobEndpoint", "secondary": "BlobSecondaryEndpoint"},
     "queue": {"primary": "QueueEndpoint", "secondary": "QueueSecondaryEndpoint"},
     "file": {"primary": "FileEndpoint", "secondary": "FileSecondaryEndpoint"},
+    "dfs": {"primary": "BlobEndpoint", "secondary": "BlobSecondaryEndpoint"},
 }
 
 
+class check_parameter_api_version(object):  # pylint: disable=client-incorrect-naming-convention
+    """Validate that the parameters passed into the current operation
+    are compatible with the current API version.
+    """
+    def __init__(self, **kwargs):
+        self.compat_map = kwargs
+
+    def __call__(self, func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            if self.compat_map and kwargs:
+                current = args[0].api_version  # client api version
+                invalid_params = [param for param, api in self.compat_map.items() if api > current]
+                invalid_values = [param for param in kwargs if param in invalid_params]
+                if invalid_values:
+                    formatted_params = "\n".join(
+                        ["'{}' (introduced in version {})".format(p, self.compat_map.get(p)) for p in invalid_values])
+                    raise ValueError(
+                        "The current API version '{}' does not support the following parameters:\n{}".format(
+                            current,
+                            formatted_params
+                        )
+                    )
+            return func(*args, **kwargs)
+        return wrapper
+
+
+class check_operation_api_version(object):  # pylint: disable=client-incorrect-naming-convention
```

Comment:
>check_operation_api_version [](start = 6, length = 27)

it seems this one is not used anywhere?

Reply:
Correct - it is not currently used anywhere.
If was going to be used for the `get_managed_disk_diff` operation.
I feel we can probably leave it there for future use.